homeassistant/components/alarm_control_panel/device_condition.py

```diff
@@ -0,0 +1,162 @@
+"""Provide the device automations for Alarm control panel."""
+from typing import Dict, List
+
+import voluptuous as vol
+
+from homeassistant.components.alarm_control_panel.const import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_CUSTOM_BYPASS,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_ARM_NIGHT,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_CONDITION,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_TYPE,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import condition, config_validation as cv, entity_registry
+from homeassistant.helpers.config_validation import DEVICE_CONDITION_BASE_SCHEMA
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from . import DOMAIN
+from .const import (
+    CONDITION_ARMED_AWAY,
+    CONDITION_ARMED_CUSTOM_BYPASS,
+    CONDITION_ARMED_HOME,
+    CONDITION_ARMED_NIGHT,
+    CONDITION_DISARMED,
+    CONDITION_TRIGGERED,
+)
+
+CONDITION_TYPES = {
+    CONDITION_TRIGGERED,
+    CONDITION_DISARMED,
+    CONDITION_ARMED_HOME,
+    CONDITION_ARMED_AWAY,
+    CONDITION_ARMED_NIGHT,
+    CONDITION_ARMED_CUSTOM_BYPASS,
+}
+
+CONDITION_SCHEMA = DEVICE_CONDITION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(CONDITION_TYPES),
+    }
+)
+
+
+async def async_get_conditions(
+    hass: HomeAssistant, device_id: str
+) -> List[Dict[str, str]]:
+    """List device conditions for Alarm control panel devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    conditions = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        state = hass.states.get(entry.entity_id)
+
+        # We need a state or else we can't populate the different armed conditions
+        if state is None:
+            continue
+
+        supported_features = state.attributes["supported_features"]
+
```

Comment:
@springstan  How about making this a bit more DRY by creating a base dict

{code}
            base_dict = {
                CONF_CONDITION: "device",
                CONF_DEVICE_ID: device_id,
                CONF_DOMAIN: DOMAIN,
                CONF_ENTITY_ID: entry.entity_id
           }
{code}

Maybe copy the dict and inserting the CONF_TYPE key or some other similar method?

Reply:
In principle I agree with you, but in this case I do not think that it is useful to change this because doing something like this
```python
base_dict.copy().update({CONF_TYPE: CONDITION_DISARMED})
```
for every condition does not make the code more readable IMHO.

Reply:
It easier for me to read since I can quickly see what is different between the lines and don't have to process each block in my head to find the most important part.  That being said, I'm not the only one reading the code so its just my opinion and certainly not a requirement.